### ISSUE TITLE
Version error when running the pipeline

### ISSUE THREAD

**Issue by maxwellevin** [tsdat/pipeline-template#24]
## Describe the bug
I'm getting a pydantic `ValidationError` that stems from an error raised by `dunamai` when it tries to get the version from git.

## Steps to Reproduce

Simply run `pytest` from the root of this repository.

If applicable, include the full traceback below:
```
==================================================================== test session starts =====================================================================
platform darwin -- Python 3.8.13, pytest-7.1.2, pluggy-1.0.0
rootdir: /Users/levi260/github/pipeline-template, configfile: setup.cfg
collected 2 items                                                                                                                                            

pipelines/example_pipeline/test/test_pipeline.py F                                                                                                     [ 50%]
test/test_pipeline_structure.py .                                                                                                                      [100%]

========================================================================== FAILURES ==========================================================================
_______________________________________________________________ test_example_pipeline_pipeline _______________________________________________________________

    def test_example_pipeline_pipeline():
        config_path = Path("pipelines/example_pipeline/config/pipeline.yaml")
>       config = PipelineConfig.from_yaml(config_path)

pipelines/example_pipeline/test/test_pipeline.py:8: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../tsdat/tsdat/config/utils.py:42: in from_yaml
    return cls(**config)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   pydantic.error_wrappers.ValidationError: 1 validation error for PipelineConfig
E   dataset -> attrs
E     Version '0.4-post.2+555d66b' does not conform to the Semantic Versioning style (type=value_error)

pydantic/main.py:331: ValidationError
================================================================== short test summary info ===================================================================
FAILED pipelines/example_pipeline/test/test_pipeline.py::test_example_pipeline_pipeline - pydantic.error_wrappers.ValidationError: 1 validation error for P...
================================================================ 1 failed, 1 passed in 5.51s =================================================================
```

## Expected behavior
I expected no error to be raised. The code version of `0.4-post.2+555d66b` seemed reasonable to me.

## Additional context
I notice a new version of `dunamai` was released 4 days ago, while the most recent commit to this repository (where the tests ran) was 6 days ago.

**Comment by maxwellevin** [comment]
Evidently not an issue when running it via GitHub Actions, as seen in #22. Interesting.

@jmcvey3 @kefeimo are you able to reproduce this error, or is it just me?

**Comment by maxwellevin** [comment]
Shoot, sorry I should have investigated a bit more before pinging you. Turns out it is just me.

At one point I had created a `v0.4` tag and release but I quickly deleted it in favor of `v0.4.0`, which is the most recent release for this repository (https://github.com/tsdat/pipeline-template/releases/tag/v0.4.0). In my clone of this repo I still had both tags, which was apparently the cause of the problem.

After running `git tag --delete v0.4` I see that the tests are passing and things are as expected.
